backend/src/api/open_source_project.ts

```diff
@@ -0,0 +1,263 @@
+import {
+  IsString,
+  isUUID,
+  IsArray,
+  IsBoolean,
+  IsUUID,
+  IsOptional,
+  IsNotEmpty,
+  IsNumber,
+  IsEnum
+} from 'class-validator';
+import {
+  Organization,
+  OpenSourceProject,
+  connectToDatabase,
+  Role,
+  ScanTask,
+  Scan,
+  User,
+  OrganizationTag,
+  PendingDomain
+} from '../models';
+import {
+  validateBody,
+  wrapHandler,
+  NotFound,
+  REGION_STATE_MAP,
+  Unauthorized
+} from './helpers';
+import {
+  isOrgAdmin,
+  isGlobalWriteAdmin,
+  isRegionalAdmin,
+  isRegionalAdminForOrganization,
+  getOrgMemberships,
+  isGlobalViewAdmin
+} from './auth';
+import { In } from 'typeorm';
+import { plainToClass } from 'class-transformer';
+import { randomBytes } from 'crypto';
+import { promises } from 'dns';
+
+/**
+ * @swagger
+ *
+ * /project/del/{projectId}/{orgId}:
+ *  delete:
+ *    description: Disconnect a particular open source project from a given organization.
+ *    parameters:
+ *      - in: path
+ *        name: id
+ *        description: Organization id
+ *    tags:
+ *    - Organizations
+ */
+export const del = wrapHandler(async (event) => {
+  const id = event.pathParameters?.orgId;
+  const projectId = event.pathParameters?.projectId;
+
+  if (!id || !projectId ) {
+    return NotFound;
+  }
+
+  if (!isGlobalWriteAdmin(event) && !getOrgMemberships(event).includes(id)) return Unauthorized;
+
+  await connectToDatabase();
+
+  try {
+    const organization = await Organization.findOne(id, { relations: ['openSourceProjects'] });
```

Comment:
Rather than querying the organization and pulling all of the open source projects (for which there could be thousands), you can directly query the open source project with the given ID, and then ensure that the given organization is part of that project.

Reply:
I tried addressing this comment, but I'm not totally sure that I understood it.

Reply:
Great -- I updated the relation to enable [cascades](https://orkhan.gitbook.io/typeorm/docs/relations#cascades), so it's only necessary to update/remove from one side.